plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -307,6 +314,20 @@ && predicateMatches(
         }
     }
 
+    public static ParquetDataSource createDataSource(
+            TrinoInputFile inputFile,
+            OptionalLong estimatedFileSize,
+            ParquetReaderOptions options,
+            AggregatedMemoryContext memoryContext,
+            FileFormatDataSourceStats stats)
+            throws IOException
+    {
+        if (estimatedFileSize.isEmpty() || estimatedFileSize.getAsLong() > options.getSmallFileThreshold().toBytes()) {
```

Comment:
consider `>=` so that we could disable the mechanism.

Reply:
File size 0 is not valid for parquet file, setting this config to 0 already allows disabling the feature